### DISCUSSION TITLE
Load animated Gif frames to MagickImageCollection

### DISCUSSION THREAD

**Discussion by tomasr78** [dlemstra/Magick.NET#1069] [Help] [answered]
How do you load animated gif frames to MagickImageCollection? I would like to load them as separated images, adjust some properties or add additional frames and save them back to gif. The code below reads only the first frame from the gif.

```
          using (var collection = new MagickImageCollection())
                {

                        collection.Add("animated.gif");
                        
                        //var image = collection.Last();
                        //image.AnimationIterations = Conversion.ToInt(ImageParameter.AnimationIterations());
                        //image.AnimationDelay = Conversion.ToInt(ImageParameter.AnimationDelay());


                    // Optionally optimize the images (images should have the same size).
                    collection.Optimize();

                    using (var fs = File.Create("result.gif"))
                    {
                        collection.Write(fs, MagickFormat.Gif);
                    }
                }
```

**Comment by tomasr78** [comment]
I have found that there is MagickImageCollection.AddRange for adding animated gifs, can I use this method to add animated and non-animated images to the collection?

**Comment by dlemstra** [comment, marked as answer]
When you do this `using (var collection = new MagickImageCollection("animated.gif"))` all the frames of the images will be added to that collection. You can then add or remove frames (images) from the collection. And with `AddRange` you can add multiple frames/images to the collection, and to add a single frame/image you will need to use the `Add` method.
